### ISSUE TITLE
PHP Warning: Array to string conversion in TYPO3 v12

### ISSUE THREAD

**Issue by timdreier** [mmunz/c1_adaptive_images#22]
PHP Warning: Array to string conversion in /app/web/vendor/typo3fluid/fluid/src/Core/ViewHelper/TagBuilder.php line 201
...

in /app/web/vendor/typo3fluid/fluid/src/Core/ViewHelper/TagBuilder.php line 201
```
if (trim((string)$attributeValue) === '' && $this->ignoreEmptyAttributes) {
    return;
}
```

at TYPO3Fluid\Fluid\Core\ViewHelper\TagBuilder->addAttribute('data-focus-area', array('x' => 0.0, 'y' => 0.0, 'width' => 0.0, 'height' => 0.0))

in /app/web/vendor/c1/c1-adaptive-images/Classes/ViewHelpers/AbstractImageBasedViewHelper.php line 291

```sql
if (!$this->tag->hasAttribute('data-focus-area')) {
    $focusArea = $cropVariantCollection->getFocusArea($cropVariant);
    if (!$focusArea->isEmpty()) {
        $this->tag->addAttribute('data-focus-area', $focusArea->makeAbsoluteBasedOnFile($image)->asArray());
    }
}
```

My be fixed by changing the affected line by the following:

```
if (!$focusArea->isEmpty()) {
    $this->tag->addAttribute('data-focus-area', json_encode($focusArea->makeAbsoluteBasedOnFile($image)->asArray()));
}
```

**Comment by mmunz** [comment]
@timdreier can you please add the fluid, how you called that viewhelper?

Edit: nevermind. i know how to test it and why i haven't found this before.

**Comment by mmunz** [comment]
I just released v1.0.2 which should fix this. Thanks for reporting.
